keps/sig-node/4815-dra-partitionable-devices/README.md

```diff
@@ -0,0 +1,2034 @@
+|                                                 7g.40gb                                               |
++------------+------------+------------+------------+------------+------------+------------+------------+
+|                       4g.20gb                     |                         X                         |
++------------+------------+------------+------------+------------+------------+------------+------------+
+|                       3g.20gb                     |                       3g.20gb                     |
++------------+------------+------------+------------+------------+------------+------------+------------+
+|          2g.10gb        |          2g.10gb        |          2g.10gb        |            X            |
++------------+------------+------------+------------+------------+------------+------------+------------+
+|          1g.10gb        |          1g.10gb        |          1g.10gb        |          1g.10gb        |
++------------+------------+------------+------------+------------+------------+------------+------------+
+|   1g.5gb   |   1g.5gb   |   1g.5gb   |   1g.5gb   |   1g.5gb   |   1g.5gb   |   1g.5gb   |     X      |
++------------+------------+------------+------------+------------+------------+------------+------------+
+| 1g.5gb+me  | 1g.5gb+me  | 1g.5gb+me  | 1g.5gb+me  | 1g.5gb+me  | 1g.5gb+me  | 1g.5gb+me  |     X      |
++------------+------------+------------+------------+------------+------------+------------+------------+
+```
+
+The most important thing to note from all of this is that partitioning a GPU
+into a set of MIG devices is not as simple as slicing it up into a set of
+fixed-size, non-overlapping partitions. Instead, one can think of a GPU as a
+"bag" of resources that can be pulled together in a discrete set of ways to
+construct a MIG device with a particular profile. When instatiating a MIG
+device with a particular profile, the set of resources available to other MIG
+devices (or the full GPU itself) gets depleted.
+
+In other words, the allocation of one MIG device may nullify the
+ability to allocate some subset of other possible MIG devices (but not all of
+them). A MIG device is considered allocatable so long as the full GPU has
+enough *unallocated* capacity to satisfy its capacity constraints across all
+dimensions. If any single dimension is unavailable, that MIG device cannot be
+allocated (but others might still be able to).
+
+For example, the following `ResourceClaim` can be used to select a set of
+non-overlapping MIG devices from a specific GPU.
+```yaml
+apiVersion: resource.k8s.io/v1alpha3
+kind: ResourceClaim
+metadata:
+  name: mig-devices
+spec:
+  spec:
+    devices:
+      requests:
+      - name: mig-1g-5gb-0
+        deviceClassName: mig.nvidia.com
+        selectors:
+        - cel:
+            expression: "device.attributes['gpu.nvidia.com'].profile == '1g.5gb'"
```

Comment:
If I specify this list in reverse order, does it mean that this won't fit on the GPU?

Reply:
Ordering doesn't matter.